packages/astro/src/cli/sync/index.ts

```diff
@@ -0,0 +1,32 @@
+import type fsMod from 'node:fs';
+import { performance } from 'node:perf_hooks';
+import { dim } from 'kleur/colors';
+import type { AstroSettings } from '../../@types/astro';
+import { error, info, LogOptions } from '../../core/logger/core.js';
+import { contentObservable, createContentTypesGenerator } from '../../content/index.js';
+import { getTimeStat } from '../../core/build/util.js';
+
+export async function sync(
+	settings: AstroSettings,
+	{ logging, fs }: { logging: LogOptions; fs: typeof fsMod }
+): Promise<0 | 1> {
+	const timerStart = performance.now();
+
+	const contentTypesGenerator = await createContentTypesGenerator({
+		contentConfigObserver: contentObservable({ status: 'loading' }),
+		logging,
+		fs,
+		settings,
+	});
+
+	try {
+		await contentTypesGenerator.init();
+	} catch (e) {
+		error(logging, 'content', 'Failed to generate content collection types: ' + e);
```

Comment:
Should new errors all now receive associated error documentation? Or is that only possible for `core`? (https://github.com/withastro/astro/blob/main/packages/astro/src/core/errors/README.md)

Reply:
To be honest, I'm note sure we document CLI errors as we would runtime errors! Let me raise in discord if that's alright.

Reply:
Update: Added `8xxx` codes for CLI errors, and added a `GenerateContentTypesError` ❤️